compliance_checker/cf/cf.py

```diff
@@ -2412,6 +2527,189 @@ def check_reduced_horizontal_grid(self, ds):
 
         return ret_val
 
+    def check_grid_mapping_attr_type(self, attr_name, attr, attr_type, variable=None):
+        """
+        Check the type of an attribute. Wrapper function, 
+        implemented for CF-1.6.
+
+        :param str attr_name: name of attribute
+        :param attr: attribute to check
+        :param str attr_type: designated type of attr; 'S', 'N', or 'D'
+        :param netCDF4.Variable variable
+        :rtype tuple
+        :return two-tuple of (bool, str|None)
+        """
+
+        # get check result, list [bool, str|None] 
+        _res = self._check_attr_type(attr, attr_type, variable)
+
+        # fully format the message if it needs it
+        if isinstance(_res[1], basestring):
+            _res[1] = _res[1].format(attr_name)
+
+        # tuple for immutability
+        return tuple(_res)
+
+    def check_grid_mapping_attr_condition(self, attr, attr_name):
+        """
+        Evaluate a condition (or series of conditions) for a particular
+        attribute. Implementation for CF-1.6.
+
+        :param attr: attribute to teset condition for
+        :param str attr_name: name of the attribute
+        :rtype tuple
+        :return two-tuple of (bool, str)
+        """
+
+        if attr_name == 'latitude_of_projection_origin':
```

Comment:
Using a `defaultdict` or similar with functions as values can be faster than large chained if/elif structure.